src/components/button/button.scss

```diff
@@ -145,6 +145,11 @@ button.md-button::-moz-focus-inner {
     &.ng-hide, &.ng-leave {
       transition: none;
     }
+
+    md-icon {
+      // Fixes icon vertical alignment.
```

Comment:
@DevVersion Agreed. I'll see if I can figure out why it actually fixes it. iirc `layout="row"` has a rule to have all children use `display: block;`.